forge-gui/res/cardsfolder/upcoming/recommission.txt

```diff
@@ -0,0 +1,7 @@
+Name:Recommission
+ManaCost:1 W
+Types:Sorcery
+A:SP$ ChangeZone | Origin$ Graveyard | Destination$ Battlefield | SubAbility$ DBPutCounter | ValidTgts$ Artifact.YouCtrl+cmcLE3,Creature.YouCtrl+cmcLE3 | TgtPrompt$ Select target artifact or creature card with mana value 3 from your graveyard | SpellDescription$ Return target artifact or creature card with mana value 3 or less from your graveyard to the battlefield. If a creature enters the battlefield this way, it enters with an additional +1/+1 counter on it.
+SVar:DBPutCounter:DB$ PutCounter | Defined$ Targeted | ConditionDefined$ Targeted | ConditionPresent$ Card.Creature | CounterType$ P1P1 | CounterNum$ 1
```

Comment:
It needs to ETB with the counter

Reply:
Hmm other similar cards like Defy Death don't have that .... I guess ETB$ True would work or does it need to be a replacement effect ?

Reply:
Rules wise,  Defy Death does make the creature enter, then puts the counter on it

if something enters unconditional with counters, then there is "WithCounter" param

for conditional counter, use AnimateSubAbility and the Counter Effect should have `ETB$ True`

Reply:
Defy Death is different, but I think we had another card like this already

I think the WithCountersType$ syntax is a bit different but that direction might work

Reply:
Ah thanks for the explanation. Forgot about animatesubability

Reply:
otherwise it might be a bit tricky and needs an replacement effect ... but @Northmoc or @Agetian  or @tool4ever could do that later ...

because if it should give a counter if a non creature artifact enters as artifact because of some static ability or clone

take this piece of code, inspired by Moritte of the Frost
```
ChangeZone: AnimateSubAbility$ DBConditionEffect

SVar:DBConditionEffect:DB$ Effect | RememberObjects$ RememberedCard | ReplacementEffects$ ETBCreat
SVar:ETBCreat:Event$ Moved | ValidCard$ Creature.IsRemembered | Destination$ Battlefield | ReplaceWith$ DBPutP1P1 | ReplacementResult$ Updated | Description$ If it's a creature, it enters with an additional +1/+1 counter on it.
SVar:DBPutP1P1:DB$ PutCounter | Defined$ ReplacedCard | CounterType$ P1P1 | ETB$ True | CounterNum$ 1 | SubAbility$ DBExile
SVar:DBExile:DB$ ChangeZone | Defined$ Self | Origin$ Command | Destination$ Exile

```

Reply:
Yeah this smells like Spark Double to me